server/tests/flight.test.js

```diff
@@ -1,13 +1,106 @@
+import mongoMock from 'mongo-mock';
 import request from 'supertest-as-promised';
 import httpStatus from 'http-status';
 import chai from 'chai';
 import { expect } from 'chai';
 import app from '../../index';
+import should from 'should';
+let MongoClient = mongoMock.MongoClient;
+MongoClient.persist = 'mongo.js';
 
-chai.config.includeStack = true;
+describe('Flights', () => {
+  let connected_db;
+  let collection;
 
-describe('#Flight ', () => {
-  describe('# GET /flight/', () => {
+  before((done)=> {
+    MongoClient.connect("mongodb://localhost:27017/gomake ", (err, db)=> {
+      connected_db = db;
+      collection = connected_db.collection("flights");
+      done();
+    });
+  });
+
+  describe('indexes', function() {
+    it('should create a unique index on callSign and flightNumber',(done)=> {
+      collection.createIndex({
+        callSign: 1,
+        flightNumber: 1
+      }, {
+        unique: true
+      }, (err, name)=> {
+        if (err) {
+          return done(err)
+        };
+        name.should.equal('callSign_1_flightNumber_1');
+        done();
+      });
+    });
+
+    it('should deny unique constraint violations on insert', function(done) {
```

Comment:
Interesting idea. That's a good idea as well. My thinking was that you have assertions on lines 56, 57, 65, 66, 76, and 77 which might benefit from being separated into three different tests (I imagine you could put 56 and 57 together, 65 and 66 togethr, and 76 and 77 together). By separating these six assertions across three tests, it might be easier to diagnose a failure in the future. If this creates a lot of additional code, you could lump these three tests inside a `describe()` block and add a `beforeEach()` block to keep setup routines that are needed in common across all three tests.